### DISCUSSION TITLE
Corner Name Circuit

### DISCUSSION THREAD

**Discussion by flaviopace** [theOehrly/Fast-F1#188] [Q&A] [answered]
First of all, awesome project. Thank you for sharing it.

Is there a way to have a dedicated "mark" for each corner?

I would have the ability to easily put name for each "corner" like: Monza Parabolica or Silverstone Stowe corner.

**Comment by theOehrly** [comment, marked as answer]
The api does not provide information about the location of corners. And detecting corners automatically is difficult. There was a bit of discussion and some code examples here: #155

**Comment by flaviopace** [comment]
Thanks. I'll take a look

**Comment by psychemedia** [comment]
Re: locating corners, I started to explore the characterisation of rally routes, including corner detection, in https://rallydatajunkie.com/visualising-rally-stages/automatically-generating-stage-descriptions.html#identifying-corners-based-on-trajr-step-gradients (see also *20.6.2 Numbering Corners*). That approach uses an R package called `trajr`; I'm not sure if there is an equivalent package in Python that could be co-opted in a similar way?

**Reply by theOehrly** [reply]
This is certainly worth a try. Before seeing how this could be done in Python, it might be best to do some validation by exporting the position data and trying to detect the corners in with the R package. If the results are any good, we can still worry about how this could be done in FastF1.
If you are interested, you could try to do this in the way that is described in the article and post the results here, so we can evaluate them.
